src/qtism/data/storage/xml/XmlDocument.php

```diff
@@ -692,21 +694,79 @@ protected function inferVersion(): string
     private function validateDocComponent(QtiComponent $docComponent): void
     {
         $branchRules = $docComponent->getComponentsByClassName('branchRule', true);
+
         if ($branchRules->count() === 0) {
             return;
         }
 
         $errors = [];
+        $components = [];
+
+        $outcomeDeclarationIds = array_map(
+            static fn (OutcomeDeclaration $outcomeDeclaration) => $outcomeDeclaration->getIdentifier(),
+            $docComponent->getComponentsByClassName('outcomeDeclaration')->getArrayCopy()
+        );
+
+        if (empty($outcomeDeclarationIds)) {
+            $errors[] = 'Outcome Declarations are required for branch rules.';
```

Comment:
This looks like a breaking change in the behavior. The presence of the outcome declarations had not been a requirement before.
Are you absolutely sure about it?

Reply:
This one will be removed (already removed from legacy code) 🙂 
Thanks!